src/process/ProcessManagerImpl.cpp

```diff
@@ -701,13 +726,23 @@ std::weak_ptr<ProcessExitEvent>
 ProcessManagerImpl::runProcess(std::string const& cmdLine, std::string outFile)
 {
     std::lock_guard<std::recursive_mutex> guard(mProcessesMutex);
+    if (!mTmpDir)
+    {
+        mTmpDir =
+            std::make_unique<TmpDir>(mApp.getTmpDirManager().tmpDir("process"));
```

Comment:
maybe you should have a `mTmpDirManager` instead of `mApp` ?
Actually - maybe we should just create the tmp dir in the constructor? that way we avoid the singleton code

Reply:
that's what i initially did, but our tests are setup in a funny way: they create an application once and potentially run newdb on the same application instance. This causes `bucket` dir to be deleted, and then re-created (see `BucketManagerImpl::dropAll`). At that point, process manager is already initialized, but its temp dir is deleted. A solution would be to spin up an application twice in our tests, once to do `newdb`, and then to actually run it. I can move tmp dir creation to the constructor if we're ok with the overhead.

Reply:
From what I see it's `ApplicationImpl::initialize` that's written in a funny way:
maybe all we need to do is move `    mProcessManager = ProcessManager::create(*this);`
 to after the newdb/upgrade block.